includes/Modules/Analytics.php

```diff
@@ -252,28 +252,39 @@ public function on_deactivation() {
 	public function get_debug_fields() {
 		$settings = $this->get_settings()->get();
 
-		return array(
-			'analytics_account_id'  => array(
+		$fields = array(
+			'analytics_dashboard_view' => array(
+				'label' => __( 'Analytics dashboard view', 'google-site-kit' ),
+				'value' => 'google-analytics-4' === $settings['dashboardView'] ? __( 'Google Analytics 4 view', 'google-site-kit' ) : __( 'Universal Analytics view', 'google-site-kit' ),
+				'debug' => $settings['dashboardView'],
+			),
+			'analytics_account_id'     => array(
 				'label' => __( 'Analytics account ID', 'google-site-kit' ),
 				'value' => $settings['accountID'],
 				'debug' => Debug_Data::redact_debug_value( $settings['accountID'] ),
 			),
-			'analytics_property_id' => array(
+			'analytics_property_id'    => array(
 				'label' => __( 'Analytics property ID', 'google-site-kit' ),
 				'value' => $settings['propertyID'],
 				'debug' => Debug_Data::redact_debug_value( $settings['propertyID'], 7 ),
 			),
-			'analytics_profile_id'  => array(
+			'analytics_profile_id'     => array(
 				'label' => __( 'Analytics view ID', 'google-site-kit' ),
 				'value' => $settings['profileID'],
 				'debug' => Debug_Data::redact_debug_value( $settings['profileID'] ),
 			),
-			'analytics_use_snippet' => array(
+			'analytics_use_snippet'    => array(
 				'label' => __( 'Analytics snippet placed', 'google-site-kit' ),
 				'value' => $settings['useSnippet'] ? __( 'Yes', 'google-site-kit' ) : __( 'No', 'google-site-kit' ),
 				'debug' => $settings['useSnippet'] ? 'yes' : 'no',
 			),
 		);
+
+		if ( ! Feature_Flags::enabled( 'ga4Reporting' ) ) {
+			unset( $fields['analytics_dashboard_view'] );
```

Comment:
Although this is how it was defined in IB, this is not a very good approach because it requires accurate maintenance in the future. There might easily happen a case when another engineer may forget to update this `unset` statement when they rename keys in the future this property won't be unset. Much better is to define that property in this `if` statement instead of adding it by default to the list.

Please, update your implementation to add the field when the feature flag is enabled. You can use the `array_merge` function to add that property to the beginning of the list.

Reply:
Thank you for your valuable feedback, @eugene-manuilov! I agree with your concern and I have updated the implementation to use `array_merge` and added the property to the beginning of the list only if the feature flag is enabled.

Just FYI, I only implemented it that way initially because it'd be easier to clean up by just removing this `if` block when removing the feature flag in the future.

Reply:
Thanks, @nfmohit.